### PR TITLE
[MAINTENANCE] Clean up `ValidationAction` API

### PR DESCRIPTION
**Pull request by cdkini**
Sets up V1's `run()` API

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/checkpoint/actions.py [hunk 1]
```diff
@@ -269,17 +270,22 @@ def _root_validate_slack_params(cls, values: dict) -> dict:
...
         expectation_suite_identifier=None,
         checkpoint_identifier=None,
+        checkpoint_result: CheckpointResult | None = None,
     ):
         logger.debug("SlackNotificationAction.run")
 
+        if checkpoint_result:
+            raise NotImplementedError(
+                f"{self.__class__.__name__} does not yet support V1 CheckpointResults."
+            )
```
Comment (cdkini) [inline great_expectations/checkpoint/actions.py]: Going to write subsequent PRs to update each of these actions and their respective renderers to work with a checkpoint result

File: tests/actions/test_core_actions.py [hunk 2]
```diff
@@ -827,3 +809,74 @@ class Config:
         parsed_action = DummyClassWithActionChild.parse_raw(json_dict)
 
         assert isinstance(parsed_action.action, action_class)
+
+
+class TestV1ActionRun:
```
Comment (cdkini) [inline tests/actions/test_core_actions.py:814]: This class is just keeping me honest as I chip away at these one-by-one. I could also just make a list and not write these tests but I like this?

File: great_expectations/checkpoint/actions.py [hunk 3]
```diff
@@ -102,11 +103,10 @@ def _using_cloud_context(self) -> bool:
         return project_manager.is_using_cloud()
 
     @public_api
-    def run(  # noqa: PLR0913
+    def run(
         self,
         validation_result_suite: ExpectationSuiteValidationResult,
         validation_result_suite_identifier: Union[ValidationResultIdentifier, GXCloudIdentifier],
-        data_asset=None,
```
Comment (cdkini) [inline great_expectations/checkpoint/actions.py:109]: Never used so just deleting

File: great_expectations/checkpoint/actions.py [hunk 4]
```diff
@@ -165,6 +161,14 @@ class (for open source Great Expectations) or the GeCloudIdentifier (from Great
         """  # noqa: E501
         return NotImplementedError
 
+    # NOTE: To be promoted to 'run' after V1 development
+    def v1_run(self, checkpoint_result: CheckpointResult) -> None:
+        return self._v1_run(checkpoint_result=checkpoint_result)
+
+    # NOTE: To be promoted to '_run' after V1 development
+    def _v1_run(self, checkpoint_result: CheckpointResult):
+        raise NotImplementedError
```
Comment (cdkini) [inline great_expectations/checkpoint/actions.py]: Can only be promoted once the legacy checkpoint is removed

Comment (cdkini) [reply great_expectations/checkpoint/actions.py]: Having this just run on the total `CheckpointResult` - makes the API much much simpler and contains all the data we need (validation results, identifiers, suite and checkpoint information, etc)

File: great_expectations/checkpoint/actions.py (hunk 3 above)
Comment (cdkini) [reply great_expectations/checkpoint/actions.py:109]: I guess I didn't need to touch this if we're removing the method entirely 😕 ?

File: great_expectations/checkpoint/actions.py [hunk 5]
```diff
@@ -165,6 +161,14 @@ class (for open source Great Expectations) or the GeCloudIdentifier (from Great
         """  # noqa: E501
         return NotImplementedError
 
+    # NOTE: To be promoted to 'run' after V1 development
```
Comment (tyler-hoffman) [inline great_expectations/checkpoint/actions.py]: IMO we should add a jira ticket and reference it here, but non-blocking.

File: great_expectations/checkpoint/actions.py [hunk 6]
```diff
@@ -165,6 +161,14 @@ class (for open source Great Expectations) or the GeCloudIdentifier (from Great
...
+    def v1_run(self, checkpoint_result: CheckpointResult) -> None:
+        return self._v1_run(checkpoint_result=checkpoint_result)
```
Comment (tyler-hoffman) [inline great_expectations/checkpoint/actions.py]: Will this method ever do more than directly wrap the private method? If not, why even have the private method?

**Review by tyler-hoffman — APPROVED** [review]

File: tests/actions/test_core_actions.py (hunk 2 above)
Comment (tyler-hoffman) [reply tests/actions/test_core_actions.py:814]: TBH I don't have strong opinions, but I'll give a weak one: the way these tests are written tells me that we want calling `v1_run` to raise. getting rid of the expectation to raise and marking the tests with an `xfail` indicates that the ideal state is to blow away the xfail in the PR that implements v1_run

**Pull request merged**